src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -442,30 +444,77 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 				if (this._isCellStyledLikeGhostText(cell)) {
 					ghostTextIndex = cursorIndex + potentialGhostIndexOffset;
 					break;
-				} else if (!this._cellStylesMatch(cursorCell, cell)) {
-					const ghostTextCell = cell;
-					// Verify that all chars for the rest of the line match this
-					// ghost text style
-					while (x < line.length) {
-						const cell = line.getCell(x++);
-						if (!cell || cell.getCode() === 0) {
-							break;
-						}
+				}
 
-						if (!this._cellStylesMatch(ghostTextCell, cell)) {
-							return -1;
-						}
+				potentialGhostIndexOffset += cell.getChars().length;
+			}
+		}
+
+		// Ghost text may not be italic or dimmed, but will have a different style than the
+		// rest of the line that precedes it.
+		if (ghostTextIndex === -1 && this.value.length > this._cursorIndex) {
```

Comment:
This section could live in a `_scanForGhostTextAdvanced` to better segment the code and help with readability.